ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/LogSegment.java

```diff
@@ -145,6 +147,10 @@ static LogSegment loadSegment(RaftStorage storage, File file, long start, long e
     });
     LOG.info("Successfully read {} entries from segment file {}", entryCount, file);
 
+    if (isOpen && end == INVALID_LOG_INDEX) {
```

Comment:
If the segment is open, the end should always be INVALID_LOG_INDEX?
Are there cases where it is not INVALID_LOG_INDEX ?

Reply:
Also how is the segment's end index initialized after a node restart ?

Reply:
Yes it should always be INVALID_LOG_INDEX. I will remove that check.

Segment endIndex is updated during readSegmentFile call. It in turn calls segment.append for each entry and updates the endIndex.